pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/MultiValueBlock.java

```diff
@@ -47,6 +50,9 @@ public BlockDocIdSet getBlockDocIdSet() {
 
   @Override
   public BlockValSet getBlockValueSet() {
```

Comment:
@Jackie-Jiang , I am concerned about the invocation of this in multi-threaded scenario. From the code, it doesn't look like it will happen since each query will create it's own. Thoughts?

Reply:
@siddharthteotia There is no multi-threading for the blocks. All the blocks are processed by one thread